qmarkdowntextedit.cpp

```diff
@@ -304,6 +304,12 @@ bool QMarkdownTextEdit::eventFilter(QObject *obj, QEvent *event) {
             _searchWidget->doSearch(
                     !keyEvent->modifiers().testFlag(Qt::ShiftModifier));
             return true;
+        } else if ((keyEvent->key() == Qt::Key_Z) &&
+                   (keyEvent->modifiers().testFlag(Qt::ControlModifier)) &&
+                   (keyEvent->modifiers().testFlag(Qt::ShiftModifier))) {
+            //handle this before undo otherwise Ctrl+Shift+Z is caught by Ctrl+Z for some reason
+            QPlainTextEdit::redo();
+            return true;
         } else if ((keyEvent->key() == Qt::Key_Z) &&
                    (keyEvent->modifiers().testFlag(Qt::ControlModifier))) {
```

Comment:
Done.

Reply:
No, I meant reverting all your PR code and instead of:

```cpp
} else if ((keyEvent->key() == Qt::Key_Z) &&
             (keyEvent->modifiers().testFlag(Qt::ControlModifier))) {
```

do a

```cpp
} else if ((keyEvent->key() == Qt::Key_Z) &&
             !keyEvent->modifiers().testFlag(Qt::ShiftModifier)
             keyEvent->modifiers().testFlag(Qt::ControlModifier)) {
```

Reply:
Ahh yes, that's even better. Will do.